### PR TITLE
Log requirements

### LINKED ISSUE #690: runtime test solution
I as a tester want to know how to test my game and for the game to tell me when something considerd "not ok" has happened

### PR DESCRIPTION
**Pull request by NinjaDanz3r**
Resolves #690

### CODE REVIEW

File: src/Engine/Manager/SoundManager.cpp [hunk 1]
```diff
@@ -126,7 +126,9 @@ void SoundManager::Update(float deltaTime) {
         processedSamples = new float[numSamples] {0};
 
     Pa_WriteStream(stream, processedSamples, *numProcessedSamples);
-    delete[] processedSamples;
+
+    if(*numProcessedSamples != 0)
```
Comment (NinjaDanz3r) [inline src/Engine/Manager/SoundManager.cpp]: INFO keyword 'if' not followed by a single space [rule](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT008)

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 3 issues
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [SoundManager.cpp#L164](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/64f9d94ed824d8e624dde51dd4e506f7a5ebd25d/src/Engine/Manager/SoundManager.cpp#L164): The function 'GetListeners' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 3 issues
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [SoundManager.cpp#L164](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/80f902c6f49059f67da007a589c8e03ddb3eb516/src/Engine/Manager/SoundManager.cpp#L164): The function 'GetListeners' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [SoundManager.cpp#L164](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/2ee968ad248b3b3e8ff5a7847b668ff026bfda9a/src/Engine/Manager/SoundManager.cpp#L164): The function 'GetListeners' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [SoundManager.cpp#L164](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/ecc87d205bcb4297333d26c685ea0facb367925f/src/Engine/Manager/SoundManager.cpp#L164): The function 'GetListeners' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/Engine/Manager/ProfilingManager.hpp [hunk 2]
```diff
@@ -87,6 +86,10 @@ class ProfilingManager {
          * @return The name.
          */
         ENGINE_API static std::string TypeToString(Type type);
+
+        ENGINE_API unsigned int MeasureRAM();
```
Comment (Chainsawkitten) [inline src/Engine/Manager/ProfilingManager.hpp]: Undocumented public method

File: src/Engine/Manager/ProfilingManager.hpp [hunk 3]
```diff
@@ -87,6 +86,10 @@ class ProfilingManager {
...
+
+        ENGINE_API unsigned int MeasureVRAM();
```
Comment (Chainsawkitten) [inline src/Engine/Manager/ProfilingManager.hpp]: Undocumented public method

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [SoundManager.cpp#L164](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c4005fb9e0c7588b1502c9952637077dd91c78bd/src/Engine/Manager/SoundManager.cpp#L164): The function 'GetListeners' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**